crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -501,17 +524,40 @@ impl RoomEventCache {
         None
     }
 
+    /// Try to find an event by id in this room, along with all relations.
+    pub async fn event_with_relations(
```

Comment:
So here's a design question: a related event could also have related events (e.g. an edit could be redacted). Should the function return the transitive closure of relations (i.e. all the related events, and all their related events recursively)? I think so, in which case we should likely implement some kind of fixed point algorithm to get all the related events.

If you agree with this and implement it, can you add a test exhibiting this use case, please?

Reply:
I think it makes sense, yes. Let's see how hard to implement this is.

Reply:
Just to confirm, there's no way relationships can be circular, right?

Reply:
This was handled in https://github.com/matrix-org/matrix-rust-sdk/pull/3870/commits/748625a8600d16d246a94958539a1602d451393c, I added a check to try to avoid circular relationships in case there could be some.